app-rating/src/main/java/com/stepstone/apprating/AppRatingDialogFragment.kt

```diff
@@ -0,0 +1,150 @@
+        if (data!!.titleColorResId != 0) {
+            dialogView.setTitleColor(data!!.titleColorResId)
+        }
+        if (data!!.contentColorResId != 0) {
+            dialogView.setContentColor(data!!.contentColorResId)
+        }
+    }
+
+    private fun setupTitleAndContentMessages(dialogView: AppRatingDialogView) {
+        if (!TextUtils.isEmpty(title)) {
+            dialogView.setTitleText(title!!)
+        }
+        if (!TextUtils.isEmpty(content)) {
+            dialogView.setContentText(content!!)
+        }
+    }
+
+    private fun setupNegativeButton(builder: AlertDialog.Builder) {
+        if (!TextUtils.isEmpty(negativeButtonText)) {
+            builder.setNegativeButton(negativeButtonText) { dialog, which -> data!!.negativeButtonClickedListener.onClicked() }
+        }
+    }
+
+    private fun setupPositiveButton(dialogView: AppRatingDialogView, builder: AlertDialog.Builder) {
+        if (!TextUtils.isEmpty(positiveButtonText)) {
+            builder.setPositiveButton(positiveButtonText) { dialog, which ->
+                val rateNumber = dialogView.rateNumber.toInt()
+                val comment = dialogView.comment
+                data!!.positiveButtonClickedListener.onClicked(rateNumber, comment)
+            }
+        }
+    }
+
+    private val title: String?
+        get() {
+            if (TextUtils.isEmpty(data!!.title)) {
+                if (data!!.titleResId == 0) {
+                    return null
+                }
+                return getString(data!!.titleResId)
+            }
+            return data!!.title
+        }
+
+    private val content: String?
+        get() {
+            if (TextUtils.isEmpty(data!!.content)) {
+                if (data!!.contentResId == 0) {
+                    return null
+                }
+                return getString(data!!.contentResId)
+            }
+            return data!!.content
+        }
+
+    private val positiveButtonText: String?
+        get() {
+            if (TextUtils.isEmpty(data!!.positiveButtonText)) {
+                if (data!!.positiveButtonTextResId == 0) {
+                    return null
+                }
+                return getString(data!!.positiveButtonTextResId)
+            }
+            return data!!.positiveButtonText
+        }
+
+    private val negativeButtonText: String?
+        get() {
+            if (TextUtils.isEmpty(data!!.negativeButtonText)) {
+                if (data!!.negativeButtonTextResId == 0) {
+                    return null
+                }
+                return getString(data!!.negativeButtonTextResId)
+            }
+            return data!!.negativeButtonText
+        }
+
+    fun setData(data: AppRatingDialog.Builder.Data) {
+        this.data = data
+    }
+
+    companion object {
+
+        fun newInstance(data: AppRatingDialog.Builder.Data): AppRatingDialogFragment {
+            val fragment = AppRatingDialogFragment()
+            fragment.setData(data)
```

Comment:
Shouldn't this be passed in as arguments Bundle in case the fragment gets destroyed and recreated later?